### PR TITLE
core: Unwrap CompletionExceptions

### PR DESCRIPTION
**Pull request by bpon**
When an unchecked exception is thrown in a completion stage handler, the exception is wrapped in a java.util.concurrent.CompletionException.

### CODE REVIEW

File: yarpc-core/src/main/java/io/yarpc/core/transport/exceptions/RequestEncodingException.java [hunk 1]
```diff
@@ -73,6 +73,7 @@ public static RequestEncodingException newMismatch(
       TransportRequest request, String procedureEncoding) {
     return fromTransportRequest(
         request,
+        procedureEncoding,
```
Comment (abhinav) [inline yarpc-core/src/main/java/io/yarpc/core/transport/exceptions/RequestEncodingException.java]: Good find. Although this change looks unrelated to this PR.

Comment (bpon) [inline yarpc-core/src/main/java/io/yarpc/core/transport/exceptions/RequestEncodingException.java]: Oh yeah, sorry.  This should have gone into #280.  Juggling too many PRs!
